javav2/example_code/scheduler/src/main/java/com/example/eventbrideschedule/scenario/EventbridgeSchedulerScenario.java

```diff
@@ -0,0 +1,365 @@
+                    executing a specific task or workflow at a predetermined time, without the need for recurring
+                    or complex scheduling.
+                    """);
+                waitForInputToContinue();
+                createOneTimeSchedule();
+                logger.info(DASHES);
+
+                logger.info("3. Create a reoccurring schedule.");
+                logger.info("""
+                    A recurring schedule is a feature that allows you to schedule and manage the execution
+                    of your serverless applications or workloads on a recurring basis. For example, 
+                    with EventBridge Scheduler, you can create custom schedules for your AWS Lambda functions, 
+                    AWS Step Functions, and other supported event sources, enabling you to automate tasks and 
+                    workflows without the need for complex infrastructure management. T
+                    """);
+                waitForInputToContinue();
+                createRecurringSchedule();
+                logger.info(DASHES);
+            }
+        } catch (Exception ex) {
+            logger.info("There was a problem with the workflow {}, initiating cleanup...", ex.getMessage());
+            cleanUp();
+        }
+
+        logger.info(DASHES);
+        logger.info("4. Clean up the resources.");
+        logger.info("Do you want to delete these AWS resources (y/n) ?");
+        String delAns = scanner.nextLine().trim();
+        if (delAns.equalsIgnoreCase("y")) {
+            cleanUp();
+        } else {
+            logger.info("The AWS resources will not be deleted.");
+        }
+        logger.info("Amazon EventBridge Scheduler workflow completed.");
+        logger.info(DASHES);
+    }
+
+    /**
+     * Cleans up the resources associated with the EventBridge scheduler.
+     * If any errors occur during the cleanup process, the corresponding error messages are logged.
+     */
+    public static void cleanUp() {
+        logger.info("First, delete the schedule group.");
+        waitForInputToContinue();
+        try {
+            eventbridgeActions.deleteScheduleGroupAsync(scheduleGroupName).join();
+
+        } catch (CompletionException ce) {
+            Throwable cause = ce.getCause();
+            if (cause instanceof SchedulerException schedulerException) {
+                logger.error("Scheduler error occurred: Error message: {}, Error code {}",
+                    schedulerException.getMessage(), schedulerException.awsErrorDetails().errorCode(), schedulerException);
+            } else {
+                logger.error("An unexpected error occurred: {}", cause.getMessage());
+            }
+            return;
+        }
+        logger.info("Next, delete the schedules");
+        waitForInputToContinue();
+        try {
+            eventbridgeActions.deleteScheduleAsync(recurringScheduleName, scheduleGroupName).join();
```

Comment:
These deleteSchedule calls are not necessary.
Deleting the group deletes the schedules in the group.
https://docs.aws.amazon.com/scheduler/latest/UserGuide/managing-schedule-group-delete.html

Reply:
Removed this logic from cleanup(). Followed the pattern in .NET